react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+export default class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { password: '' };
+    this.handlePassword = this.handlePassword.bind(this);
+  }
+
+  handlePassword(event) {
+    this.setState({ password: event.target.value });
+  }
+
+  render() {
+    let symbol;
+    let text;
+    let view;
+
+    if (this.state.password.length === 0) {
+      symbol = 'fa-sharp fa-solid fa-xmark red';
+      view = 'visible';
+      text = 'A password is required.';
+    } else if (this.state.password.lenght !== 0 && this.state.password.length < 8) {
+      symbol = 'fa-sharp fa-solid fa-xmark red';
+      view = 'visible';
+      text = 'Your password is too short.';
+    } else if (this.state.password.length >= 8) {
```

Comment:
Just an `else` here would be fine. If it makes it past the first 2 checks it is guaranteed that the password is 8 characters or more. There is no need to check the opposite of what has already been checked.